server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/internal/NettyCatsRequestBody.scala

```diff
@@ -10,33 +10,48 @@ import io.netty.buffer.ByteBufUtil
 import io.netty.handler.codec.http.{FullHttpRequest, HttpContent}
 import sttp.capabilities.fs2.Fs2Streams
 import sttp.tapir.model.ServerRequest
-import sttp.tapir.server.interpreter.{RawValue, RequestBody}
+import sttp.tapir.server.interpreter.{RawValue, RequestBody, RequestBodyToRawException}
 import sttp.tapir.{FileRange, InputStreamRange, RawBodyType, TapirFile}
 
 import java.io.ByteArrayInputStream
 import java.nio.ByteBuffer
+import sttp.tapir.DecodeResult
 
 private[netty] class NettyCatsRequestBody[F[_]](createFile: ServerRequest => F[TapirFile])(implicit val monad: Async[F])
     extends RequestBody[F, Fs2Streams[F]] {
 
   override val streams: Fs2Streams[F] = Fs2Streams[F]
 
-  override def toRaw[R](serverRequest: ServerRequest, bodyType: RawBodyType[R]): F[RawValue[R]] = {
+  override def toRaw[R](serverRequest: ServerRequest, bodyType: RawBodyType[R], maxBytes: Option[Long]): F[RawValue[R]] = {
 
+    def nettyRequestBytes: F[Array[Byte]] = serverRequest.underlying match {
+      case req: FullHttpRequest =>
+        val buf = req.content()
```

Comment:
do we have any control over whether we get a `FullHttpRequest` or not? Maybe I misunderstand something, but doesn't calling `req.content()` mean that the request data is fully read into memory anyway? Maybe we should always use streaming, if we want to avoid reading the entire request

Reply:
Netty Future and Id backends use `NettyConfig.defaultNoStreaming`, which builds the pipeline without reactive streams handler. These backends will deal only with `FullHttpRequest`.
If we wanted to make Netty streaming-only underneath, we'd need a way to read `StreamedHttpRequest` (with `Publisher[HttpData]` underneath) into raw values. This would require some concrete implementation of reactive streams. Can this be written manually just for our purposes? Some kind of blocking naive Subscriber? Only to end up with what we have now with `FullHttpRequest`.

Reply:
well a limited subscriber, which would error if there's too much data coming in. Take a look at [`SimpleSubscriber`](https://github.com/softwaremill/sttp/blob/e7ed7720a3ae4389c4b755b476e94bafb26ba694/async-http-client-backend/src/main/scala/sttp/client4/asynchttpclient/reactive.scala#L18), maybe this will be helpful

Reply:
Thanks, this `SimpleSubscriber` looks exactly what we need to get rid of `NettyConfig.defaultNoStreaming`. I guess this could be a separate improvement, implemented in another PR, shall I create an issue for this?

Reply:
well  ... can you properly implement maxLength for raw values (that is, no more than +/- maxLength bytes are read into the server's memory) without it?

Reply:
I thought of just making the Id/Future backends "not supported" with this feature, but maybe it's indeed better to add the subscriber already in this PR and cover all netty backends in one go.

Reply:
will this case ever happen now, after the streaming changes?

Reply:
No, generally `NettyCatsRequestBody` is now finished yet, it will be used only for the Id server. Anything regarding Id server is still TODO.